### PR TITLE
feat: validation and test for rest of fields

### PR DESCRIPTION
**Pull request by Jacksonzhang34**
**Issue #, if available:**

**Description of changes:**

**Why is this change necessary:**
wizard input checker for build option, publish configs, and gdk_version
**How was this change tested:**
Tested through unit testing 
**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gdk/wizard/WizardChecker.py [hunk 1]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
             return False
         except ValueError:
             return len(input_value) > 0
+
+    def check_build_options(self, input_value):
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:87]: rename these `check_`* methods to `is_valid_*`.

File: gdk/wizard/WizardChecker.py [hunk 2]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+        if len(input_obj) > 2:
+            return False
+
+        if "excludes" not in input_obj or "zip_name" not in input_obj:
+            return False
+
+        excludes = input_obj.get("excludes")
+        if (
+            not isinstance(excludes, list)
+            or len(excludes) < 1
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:106]: Nit: Customers can specify empty list for `excludes`.

File: gdk/wizard/WizardChecker.py [hunk 3]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+            or not all(isinstance(item, str) for item in excludes)
+        ):
+            return False
+
+        zip_name = input_obj.get("zip_name")
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:111]: If `zip_name` does not exist, it is returned as `None`.
Validate only when `zip_name` exists.

File: gdk/wizard/WizardChecker.py [hunk 4]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+        try:
+            # Convert the input string to a dictionary (object)
+            input_obj = json.loads(input_value)
+        except json.JSONDecodeError:
+            return False
+
+        if not isinstance(input_obj, dict):
+            return False
+
+        if len(input_obj) > 2:
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:97]: It is ok if customers want to specify an empty object for build options. ie. {} is still a valid input IMO.

File: gdk/wizard/WizardChecker.py [hunk 5]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+            return False
+
+        if "excludes" not in input_obj or "zip_name" not in input_obj:
+            return False
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:101]: If customer wants to include random keys other than these, it is still fine. So you can remove this check as well.

File: gdk/wizard/WizardChecker.py (hunk 2 above)
Comment (Jacksonzhang34) [reply gdk/wizard/WizardChecker.py:106]: I saw that on the schema it says : 
"options": {
                                                "type": "object",
                                                "description": "configuration options for the zip build system",
                                                "properties": {
                                                    "excludes": {
                                                        "type": "array",
                                                        "description": "regex patterns of files to exclude",
                                                        "minItems": 1
                                                    },
                                                    "zip_name": {
                                                        "type": "string"
                                                    }
                                                }
                                            }

Comment (Jacksonzhang34) [reply gdk/wizard/WizardChecker.py:106]: so i thought the minitem was at least one

File: gdk/wizard/WizardChecker.py (hunk 1 above)
Comment (Jacksonzhang34) [reply gdk/wizard/WizardChecker.py:87]: got it, changing now

File: gdk/wizard/WizardChecker.py [hunk 6]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+
+        excludes = input_obj.get("excludes")
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:103]: Validate only when "excludes" is present.

File: gdk/wizard/WizardChecker.py (hunk 2 above)
Comment (saranyailla) [reply gdk/wizard/WizardChecker.py:106]: Got it. You can change it to empty list here.

File: gdk/wizard/WizardChecker.py [hunk 7]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+
+        return True
+
+    def check_bucket(self, input_value):
+        # input must be a non-empty string
+        return isinstance(input_value, str) and len(input_value) > 0
+
+    def check_region(self, input_value):
+        # input must be a non-empty string
+        return isinstance(input_value, str) and len(input_value) > 0
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:123]: This is for later. This must be one of the AWS regions that `Greengrass` supports. Eg. `us-east-1 `

File: gdk/wizard/WizardChecker.py [hunk 8]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+
+        if not isinstance(input_object, dict):
+            return False
+
+        if "file_upload_args" in input_object and isinstance(
+            input_object["file_upload_args"], dict
+        ):
+            return True
+
+        return False
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:141]:
```suggestion
        _file_upload_args = input_object.get("file_upload_args",{})
        if not isinstance(_file_upload_args, dict):
            return False

        return True
```

File: gdk/wizard/WizardChecker.py [hunk 9]
```diff
@@ -77,3 +83,70 @@ def is_valid_custom_build_command(self, input_value):
...
+    def check_gdk_version(self, input_value):
+        gdk_version_pattern = (
+            "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)"
+            "(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\."
+            "(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?"
+            "(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?"
+        )
+        if re.match(gdk_version_pattern, input_value):
+            return True
+        return False
```
Comment (saranyailla) [inline gdk/wizard/WizardChecker.py:152]:
```suggestion
        return re.match(gdk_version_pattern, input_value)
```

File: gdk/wizard/WizardChecker.py (hunk 8 above)
Comment (Jacksonzhang34) [reply gdk/wizard/WizardChecker.py:139]: accepted this change

File: gdk/wizard/WizardChecker.py (hunk 9 above)
Comment (Jacksonzhang34) [reply gdk/wizard/WizardChecker.py:150]: I tried this at first it doesn't work unfortunately

**Review by saranyailla — APPROVED** [review]

**Review by urvashijain18 — APPROVED** [review]

**Pull request merged**